pkg/docker/config/config_test.go

```diff
@@ -675,3 +676,170 @@ func TestAuthKeysForRef(t *testing.T) {
 		require.Equal(t, tc.expected, result, tc.name)
 	}
 }
+
+func TestSetCredentials(t *testing.T) {
+	for _, tc := range []struct {
+		registries []string
+		assert     func(dockerConfigFile)
+	}{
+		{
+			registries: []string{"quay.io"},
+			assert: func(auth dockerConfigFile) {
+				require.Len(t, auth.AuthConfigs, 1)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io"].Auth)
+			},
+		},
+		{
+			registries: []string{"quay.io/a/b/c/d/image"},
+			assert: func(auth dockerConfigFile) {
+				require.Len(t, auth.AuthConfigs, 1)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a/b/c/d/image"].Auth)
+			},
+		},
+		{
+			registries: []string{
+				"quay.io/a/b/c",
+				"quay.io/a/b",
+				"quay.io/a",
+				"quay.io",
+				"my-registry.local",
+				"my-registry.local",
+			},
+			assert: func(auth dockerConfigFile) {
+				require.Len(t, auth.AuthConfigs, 5)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a/b/c"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a/b"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["my-registry.local"].Auth)
+			},
+		},
+	} {
+		tmpFile, err := ioutil.TempFile("", "auth.json.set")
+		require.NoError(t, err)
+		_, err = tmpFile.WriteString("{}")
+		require.NoError(t, err)
+		defer os.RemoveAll(tmpFile.Name())
+		sys := &types.SystemContext{AuthFilePath: tmpFile.Name()}
+
+		for _, registry := range tc.registries {
+			_, err := SetCredentials(sys, registry, "username", "password")
+			require.NoError(t, err)
+		}
+
+		auth, err := readJSONFile(tmpFile.Name(), false)
+		require.NoError(t, err)
+
+		tc.assert(auth)
+	}
+}
+
+func TestRemoveAuthentication(t *testing.T) {
+	testAuth := dockerAuthConfig{Auth: "ZXhhbXBsZTpvcmc="}
+	for _, tc := range []struct {
+		config     dockerConfigFile
```

Comment:
(Absolutely non-blocking: This could be a `preexistingKeys []string`. OTOH it would not be that much of a win.)

Reply:
Decided to leave the assert function to have more flexibility for test additions.